r2dbc-mysql/src/main/java/JasyncRow.kt

```diff
@@ -0,0 +1,36 @@
+package com.github.jasync.r2dbc.mysql
+
+import com.github.jasync.sql.db.RowData
+import io.r2dbc.spi.Row
+
+
+class JasyncRow(private val rowData: RowData) : Row {
+
+
+    override fun <T> get(identifier: Any, requestedType: Class<T>): T? {
+
+
+        if (identifier is String) {
```

Comment:
Initially dropping the requested type (`Class<T>`) is okay, but make sure to consider the requested type at some point in time as this is part of the SPI contract, otherwise, the calling side will run into `ClassCastException`.

Reply:
I'm not sure I understand the spi. should the returned value be of type `requestedType`?

Reply:
Exactly. Each database type maps to a specific Java type (e.g. `TINYINT` to `Byte`, `SMALLINT` to `Short`). You should be able to retrieve a `TINYINT` as `Integer.class` or as `Long.class`. The caller requests the type `Integer.class` and expects the `get(Any, Class<T>)` method to return that specific type.

This conversion makes only sense for convertible types (`Numbers`, Dates (`LocalDateTime` to `LocalTime` or `LocalDate`) and converting values to `String`).

For a start, if you don't want to implement an advanced Codec mechanism, it would make sense to consider the mentioned three conversions and if the value is not `null`, to `requestedType.cast(…)` the value.

In contrast,  `get(Any)` returns the native type (i.e. what happens right now, that the driver determines the most appropriate type).

Reply:
as you mentioned the driver has "builtin" types right now depending on the db type. I am not what you mean to do for a start? just a cast? didn't understand the different cases you mentioned.

Reply:
The method should do at least a cast. 

I would also suggest adding the following conversion rules:
* If a value is a `Number` and the requested type is a number, too, then convert the value to `T`  using `Number.intValue()`, `Number.longValue()` methods.
* If the requested type is `String`, then call `toString()` on the value.
* If a value is `LocalDate`, `LocalDateTime`, or `LocalTime` and the requested type is one of these date types, then convert the value into the requested value using the provided methods on `LocalDate` etc (e.g. `LocalDateTime.atStartOfDay()`).

Does this make sense?

Reply:
yes, I added it, please review again to see we are aligned. Note that for DateTime the lib is using joda so I had to do an extra conversion.

Reply:
Looks pretty decent now. I wasn't aware that you build on top of Joda. For R2DBC we want to stay within the Java 8 type realm for primitives. The native `get(Any)` method should ideally return Java 8 types for dates. Retrieving a Joda type would be still possible via `get(Any, Class<T>)` when requesting the Joda `LocalDateTime`.

Reply:
I will change that. we have joda as a legacy from the original driver.